### ISSUE TITLE
Nested setting disappears after validation

### ISSUE THREAD

**Issue by fswane** [dynaconf/dynaconf#974]
**Describe the bug**
When two nested settings with the same name (in the last part) are validated, the one setting is no longer present after validation.

**To Reproduce**

This is best demonstrated with a test case:

```
        from dynaconf import Validator, Dynaconf
        settings = Dynaconf()

        # Register a single validator.
        settings.validators.register(Validator('foo.bar.value', must_exist=True, is_type_of=bool))

        # Add two nested settings.
        settings.set("foo.value", True)
        settings.set("foo.bar.value", True)

        # Check that both settings are present.
        assert settings.foo.bar.value == True
        assert settings.foo.value == True

        # Validate the settings.
        settings.validators.validate()

        # Check that both settings are still present.
        assert settings.foo.bar.value == True
        assert settings.foo.value == True   # ERROR:  This line fails: "'DynaBox' object has no attribute 'value'"
```
The very last line of the test fails with an exception:

```
     dynaconf.vendor.box.exceptions.BoxKeyError: "'DynaBox' object has no attribute 'value'"

../../../.venv/lib/python3.8/site-packages/dynaconf/vendor/box/box.py:175: BoxKeyError
```

**Expected behavior**
It is expected that both settings are still present after validation.

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Python 3.8
 - Dynaconf Version 3.2.0  and 3.1.12   
 
 - Note:  With Dynaconf 3.1.11 the behaviour is as expected, i.e. this problem is present only since version 3.1.12.

**Comment by pedro-psb** [comment]
Just as a hint for future investigation, a quick bisect showed that probably [this change](https://github.com/dynaconf/dynaconf/commit/816a50019cd79fe59a56fe21eeec30f7769d3a75#diff-32c3b833b4d892e0c8c968669eb1875096b9e7ef19024dff76f32c2771669615) introduced the bug.

**Comment by rochacbruno** [comment]
#981 will probably address this issue

**Comment by rochacbruno** [comment]
To be fixed on #1143

**Comment by fswane** [comment]
Hi.

I am still getting the exact same error as reported in August 2023.

**Environment:**

OS: Ubuntu 22.04

Python 3.1

Dynaconf Version 3.2.11

The test posted above still fails with exactly the same issue.

**Question:**

Am I doing something wrong in the test code above?
